x/delayedack/keeper/finalize_test.go

```diff
@@ -0,0 +1,120 @@
+package keeper_test
+
+import (
+	"github.com/dymensionxyz/dymension/v3/app/apptesting"
+	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+	"github.com/dymensionxyz/dymension/v3/x/delayedack/types"
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+func (s *DelayedAckTestSuite) TestFinalizePacket() {
+	rollapp := "rollapp_1234-1"
+
+	cases := []struct {
+		name          string
+		packet        commontypes.RollappPacket
+		rollappHeight uint64
+		expectErr     bool
+		errContains   string
+	}{
+		{
+			name: "success",
+			packet: commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 8,
+				Packet:      getNewTestPacket(2),
+			},
+			rollappHeight: 10,
+			expectErr:     false,
+			errContains:   "",
+		},
+		{
+			name: "error: packet not found: packet status is not pending",
+			packet: commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 8,
+				Packet:      getNewTestPacket(2),
+			},
+			rollappHeight: 10,
+			expectErr:     true,
+			errContains:   "get rollapp packet:",
+		},
+		{
+			name: "error: packet proof height is greater than rollapp height",
+			packet: commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 15,
+				Packet:      getNewTestPacket(2),
+			},
+			rollappHeight: 10,
+			expectErr:     true,
+			errContains:   "packet height is not finalized yet: height '15', latest height '10'",
+		},
+	}
```

Comment:
tests for errors should use errorsmod.IsOf, not string comparison

Reply:
ahh
I think because err.Wrapf is buggy as methon on sdk error
Need to use errorsmod.Wrapf(err,..)